src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -30,47 +32,41 @@ def execute(self, reqStatus):
             self.logger.debug('+++ monit found %s requests in %s state',
                               len(requests), reqStatus)
 
-            # FIXME: completion threshold should come either from unified
-            # or campaign configuration, e.g.
-            # thr = self.unifiedConfig().get('completedThreshold', 100)
-            thr = 100
-
             requestStatuses = {}  # keep track of request statuses
+            since = self.msConfig.get('UpdateInterval', 4*24*60*60)
             for reqName in requests:
                 req = {'name': reqName, 'reqStatus': reqStatus}
                 self.logger.debug("+++ request %s", req)
                 # obtain status records from couchdb for given request
-                transferRecords = self.getTransferRecords(reqName)
+                transferRecords = self.getTransferRecords(reqName, since)
```

Comment:
See comment I made above. Going workflow by workflow gives us the best memory footprint, but a poor cpu performance. I'd rather have the best cpu performance and an average memory footprint. Thus, let's get data in bulk when possible.